### DISCUSSION TITLE
What is the relation between theme brand colors and primaryColor?

### DISCUSSION THREAD

**Discussion by nouman91** [mantinedev/mantine#688] [Q&A] [answered]
As per documentation you can define your brand color like this
`brand: ['#F0BBDD', '#ED9BCF', '#EC7CC3', '#ED5DB8', '#F13EAF', '#F71FA7', '#FF00A1', '#E00890', '#C50E82','#AD1374' ],
primaryColor: 'brand',`

Now if I create a button with these theme configurations color #7 (#FF00A1) gets selected. 

If I am creating brand color then how can I make sure which color gets selected as a base primary color?

**Comment by rtivital** [comment, marked as answer]
It depends on component variant and color scheme, you can check how colors for variants are computed here – https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/theme/utils/get-shared-color-scheme/get-shared-color-scheme.ts

**Reply by nouman91** [reply]
Thank you for your response. Apologies, I should have been more specific, what about the **Default** variant. 

I am also going through the code you referenced.

**Reply by rtivital** [reply]
`default` variant is described here – https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/theme/utils/get-shared-color-scheme/get-shared-color-scheme.ts#L47-L54

**Reply by nouman91** [reply]
Apologies, if I am asking too many questions. I was able to map out most of the colors with variants. However, I still am not able to map out with the default color. 

My current setup.

Using mantine default color scheme
Button without any customization 
`
<MantineProvider>
<Button>
          Setting{" "}
 </Button>
 </MantineProvider>
`
The button background color is blue, which is number 6 in the blue array. As per the code snippet you shared it should be the theme. white, which is plain white color.

is there anything I am missing?

**Reply by rtivital** [reply]
It's not default variant, it's `filled`, you can find colors here – https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/theme/utils/get-shared-color-scheme/get-shared-color-scheme.ts#L112-L118

**Reply by nouman91** [reply]
Thank you so much for replying and helping me understand how color gets applied. Never seen a library owner being so much active. 

I have one further question regarding theming. 

I am building a theme for my project, and as per docs, it is advised to create 10 color variants.  While doing that, how can I make sure that a specific color variant gets applied as the base primary color to components such as buttons? For reference, let's say the following are my brand colors

`brand:['#7AD1DD', '#5FCCDB', '#44CADC', '#2AC9DE', '#1AC2D9', '#11B7CD', '#09ADC3', '#0E99AC', '#128797', '#147885'],
primaryColor:brand`

and I want **#5FCCDB** to be my primary color, what theme configurations I should be doing? 

and again thank you for your support, looking forward to your response.

**Reply by rtivital** [reply]
You can find colors alignment logic in function that I've sent earlier, if it is hard to understand, set your color at index 6

**Reply by FrancescoSaverioZuppichini** [reply]
Hi @rtivital  the links are not working :) I'd like to know which collor is used for the buttons background

**Reply by rtivital** [reply]
https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/theme/functions/fns/variant/variant.ts
